### PR TITLE
feat: add support for Yatch YDHS01

### PR DESCRIPTION
**Pull request by afonsofonbraga**
# What is this PR for

This PR contains the implementation of the Yatch YDHS01 sensor. I created it but I couldn't test it right now because I had other important things to do with jetson. 
# How I did it

# Results, How I tested
This part still needs to be tested, We was not able to get the serial number of the sensor to differentiate both of them. 

# Checklist
- [x] Assign yourself  in the PR
- [ ] Write unit tests (when relevant)
- [x] Run rubocop and rake tests locally
- [ ] If this PR initialize a CMAKE package please [enable styling and linting](https://github.com/tidewise/wetpaint-buildconf/blob/master/README.md#enabling-styling-and-linting-checks-for-a-cmake-package)
- [x] Analyse if this PR modifies the UI, if so:
    - [ ] Tested Tupan's simulation (Charts) :world_map:
    - [ ] Tested Tupan's simulation (Hud) :joystick:
    - [ ] Tested ROV's simulation :joystick:

### CODE REVIEW

File: nmea2000.orogen
Comment (rayssinha-soares) [inline nmea2000.orogen:1]: what's a 'reliable connection' in these cases?

File: tasks/YatchYDHS01Task.hpp
Comment (rayssinha-soares) [inline tasks/YatchYDHS01Task.hpp:1]: remove dead code

**Comment by doudou** [comment]
@higor-melo since i did a review in-person, waiting for the revised version to do it here ...

File: tasks/YatchYDHS01Task.cpp [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+    if (!YatchYDHS01TaskBase::startHook())
+        return false;
+    return true;
+}
+void YatchYDHS01Task::updateHook()
+{
+    YatchYDHS01TaskBase::updateHook();
+
+    Message msg;
+    while (_msg_in.read(msg) == RTT::NewData) {
```
Comment (rayssinha-soares) [inline tasks/YatchYDHS01Task.cpp:65]: why did you use `while` instead of `if` ? the task is already in loop

**Comment by higor-melo** [comment]
> @higor-melo since i did a review in-person, waiting for the revised version to do it here ...

Done it. I think i got all the changes requested. Please let me know if i forgot anything.
Also, when i removed the switch case the values started updating correctly, even when changing the order between temperature/humidity.. I made sure to try the code with the temperature PGN first and humidity still updated normally.

**Review by rayssinha-soares — APPROVED** [review]

**Review by doudou — APPROVED** [review]

**Review by DeboraTahara — APPROVED** [review]

**Pull request merged**